### PR TITLE
refector the xpt2046 so it will be a platform for the touchscreen component

### PR DESCRIPTION
**Pull request by nielsnl68**
** I Closed this PR because i messed up my last try to get my source up-to-date **

See the new PR here: PR#3793

----------------------------------------------------------------------

- removing his own binarysensor component.
- when interupt pin is set it will now react on the event.
- otherwise it will start polling as the old behavier dit.

# What does this implement/fix? 
it will now use the Touchscreen eco system as a platform component.

Quick description and explanation of changes
Before the xpt2046 could only be used as a standalone componet, missing all the futures that the Touchscreen platform has,
this is now being fixed.

The old method is fully removed. howhever there should be a error warning to expain that you should use it as platform of the touchscreen.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Other: 
All existing yaml keys are still vaild and for the binary_sensors the platform need to be updated to `touchscreen` instead of `xtp2046`

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2259

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
touchscreen:
  - platform: xpt2046
    display: my_display
    id: my_toucher
    cs_pin: 14
    #interupt_pin: 16
    report_interval: 50ms
    threshold: 400
    calibration_x_min: 3860
    calibration_x_max: 280
    calibration_y_min: 340
    calibration_y_max: 3860
    #swap_x_y: true
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by nielsnl68 — COMMENTED** [review]
not sure what this will do,

**Comment by numo68** [comment]
Hello @nielsnl68 , thanks for porting this to the touchscreen platform! - frankly I did not follow that part of ESPHome so I did not even know it is now possible. Unfortunately I won't be able to test with my setup used to develop the original xpt2046 in the near future but from a casual review the PR looks fine to me.

**Comment by nielsnl68** [comment]
> 

Did you ever did some test with the interrupt used?  I changed it a bit so it is now using the esphomes interrupt mechanic. But i was not able to test it.
Not sure of the pin will stay active as long you press the screen and move around or that it will blink per movement. (i hope the last thing)

File: esphome/components/xpt2046/__init__.py [hunk 1]
```diff
@@ -1,129 +0,0 @@
-import esphome.codegen as cg
```
Comment (jesserockz) [inline esphome/components/xpt2046/__init__.py:1]: Make sure to leave a basic `CONFIG_SCHEMA` here with `cv.invalid` letting users know to check the docs. Look for other examples in the codebase

Comment (nielsnl68) [reply esphome/components/xpt2046/__init__.py:1]: Good idea, i added it also for the old `binary_sensor` and the remove/changed keys.

**Pull request closed without merging**